src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +141,52 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            ReadOnlySpan<byte> inputValue = value;
+
+            // We will ignore the leading and trailing occurrences of the white space in the json check.
+            int start = 0;
+            while (start < inputValue.Length && char.IsWhiteSpace(Convert.ToChar(inputValue[start])))
+            {
+                start++;
+            }
+            int end = inputValue.Length - 1;
+            while (end >= start && char.IsWhiteSpace(Convert.ToChar(inputValue[end])))
```

Comment:
Why do we care to trim whitespace?  `Utf8JsonWriter` is going to collapse all whitespace and indentation by default unless we're explicitly setting the [Indented](https://docs.microsoft.com/dotnet/api/system.text.json.jsonwriteroptions.indented?view=net-6.0#system-text-json-jsonwriteroptions-indented) option.

Reply:
We are trying to do a "fast" check to see if we should call WriteJson or WriteString.  If we incorrectly call WriteString for json it comes out as `"{"foo": "bar"}"` with extra quotes around it which is not correct.  We could just try to parse with JsonElement and catch always but this is very slow.  We could also use JsonReader which is bit faster.  Instead we believe we can be 90%+ accurate with this method and therefore only incur the slow parsing fail in a small percentage of cases.  Our worst case runtime is no worse than always doing a full parse, but our average case is significantly better.